xmlwriter.py

```diff
@@ -17,126 +17,131 @@
 
     def close(self, element):
-	assert element == self.path[-1]
-	del self.path[-1]
-	self.write(self.indent_str() + u'</' + element + u'>\n')
+        assert element == self.path[-1]
+        del self.path[-1]
+        self.write(self.indent_str() + '</' + element + '>\n')
 
     def openComment(self):
-	self.write(u'<!--\n')
-	self.path.append('u<!--')
+        self.write('<!--\n')
+        self.path.append('u<!--')
 
     def closeComment(self):
-	assert self.path[-1] == 'u<!--'
-	del self.path[-1]
-	self.write(u'-->\n')
+        assert self.path[-1] == 'u<!--'
+        del self.path[-1]
+        self.write('-->\n')
 
     formatRaw = 0
     formatIndent = 1
     formatBreakLines = 2
     formatFill = 3
 
     def fillParagraph(self, w):
-	indent_str = self.indent_str()
-	ll = []
-	l = [] ; n = len(indent_str)
-	for a in w.split():
-	    if n + len(a) < self.margin:
-		n = n + len(a) + 1
-		l.append(a)
-	    else:
-		ll.append(indent_str + string.join(l))
-		l = [a] ; n = len(indent_str) + len(a)
-	if len(l) > 0:
-	    ll.append(indent_str + string.join(l))
-	return ll
-
-    def cdata(self, w, format = formatFill):
-	if 'u<!--' in self.path:
-	    w = w.replace(u'--', u'=') # comment must not contain double-hyphens
-	if format == self.formatRaw:
-	    out = [ w ]
-	elif format == self.formatIndent:
-	    indentStr = self.indent_str()
-	    out = [ indentStr + line for line in w.split(u'\n') ]
-	elif format == self.formatBreakLines:
-	    out = [ self.fillParagraph(line) for line in w.split(u'\n') ]
-	    out = reduce(operator.add, out)
-	elif format == self.formatFill:
-	    out = self.fillParagraph(w)
-	self.write(string.join(out, u'\n') + u'\n')
+        indent_str = self.indent_str()
+        ll = []
+        l = []
+        n = len(indent_str)
+        for a in w.split():
+            if n + len(a) < self.margin:
+                n = n + len(a) + 1
+                l.append(a)
+            else:
+                ll.append(indent_str + " ".join(l))
+                l = [a]
+                n = len(indent_str) + len(a)
+        if len(l) > 0:
+            ll.append(indent_str + " ".join(l))
+        return ll
+
+    def cdata(self, w, format=formatFill):
+        if 'u<!--' in self.path:
+            w = w.replace('--', '=') # comment must not contain double-hyphens
+        if format == self.formatRaw:
+            out = [w]
+        elif format == self.formatIndent:
+            indentStr = self.indent_str()
+            out = [indentStr + line for line in w.split('\n')]
+        elif format == self.formatBreakLines:
+            out = [self.fillParagraph(line) for line in w.split('\n')]
+            out = reduce(operator.add, out)
+        elif format == self.formatFill:
+            out = self.fillParagraph(w)
+        self.write("\n".join(out) + '\n')
 
     def formatted_cdata(self, s):
-	for w in string.split(s, u'\\n'):
-	    self.cdata(w, self.formatFill)
+        for w in " ".split(s, '\\n'):
```

Comment:
also suspicious -- I would assume s.split('\\n')?
Could you check your code once more to makes sure these mistakes are gone?

Reply:
I doubt I would be able to see all of it

Reply:
and by '\n' I meant r'\n' or '\\\\n' -- github eats the slashes in comments